src/test/java/io/github/zap/commons/utils/ReflectionUtilsTest.java

```diff
@@ -0,0 +1,87 @@
+package io.github.zap.commons.utils;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.AbstractList;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class ReflectionUtilsTest {
```

Comment:
since this is the only test you've written, have you made sure the other non-trivial stuff works?

Reply:
No, I made the test solely for the Reflection stuff because it is so damn hard to debug recursion. I will eventually add other tests but for now, my time is kinda run out